collector/tcp.go

```diff
@@ -0,0 +1,249 @@
+// returns data points from Win32_PerfRawData_Tcpip_TCPv4
+
+// https://msdn.microsoft.com/en-us/library/aa394341(v=vs.85).aspx (Win32_PerfRawData_Tcpip_TCPv4 class)
+
+package collector
+
+import (
+	"log"
+
+	"github.com/StackExchange/wmi"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+func init() {
+	Factories["tcp"] = NewTCPCollector
+}
+
+// A TCPCollector is a Prometheus collector for WMI Win32_PerfRawData_Tcpip_TCPv4 metrics
+type TCPCollector struct {
+	ConnectionFailures          *prometheus.Desc
+	ConnectionsActive           *prometheus.Desc
+	ConnectionsEstablished      *prometheus.Desc
+	ConnectionsPassive          *prometheus.Desc
+	ConnectionsReset            *prometheus.Desc
+	FrequencyObject             *prometheus.Desc
+	FrequencyPerfTime           *prometheus.Desc
+	FrequencySys100NS           *prometheus.Desc
+	SegmentsPerSec              *prometheus.Desc
```

Comment:
*PerSec is actually a misleading name in WMI - these count the total number, not the per-second rate (The reason for the name is probably that they can be _used_ to calculate the rate). So replacing "PerSec" with "Total" is better here.